api/reviews_api.py

```diff
@@ -115,3 +126,38 @@ def do_get(self, **kwargs) -> dict[str, Any]:
         'gates': dicts,
         'possible_assignee_emails': possible_assignees_by_gate_type
         }
+
+  def do_post(self, **kwargs) -> dict[str, str]:
+    """Set the assignees for a gate."""
+    user, fe, gate, feature_id, gate_id = get_user_feature_and_gate(
+        self, kwargs)
+    assignees = self.get_param('assignees')
+
+    self.require_permissions(user, fe, gate)
+    self.validate_assignees(assignees, fe, gate)
+    old_assignees = gate.assignee_emails
+    gate.assignee_emails = assignees
+    gate.put()
+    notifier_helpers.notify_assignees(
+        fe, gate, user.email(), old_assignees, assignees)
+
+    # Callers don't use the JSON response for this API call.
+    return {'message': 'Done'}
+
+  def require_permissions(self, user, feature, gate):
+    """Abort the request if the user lacks permission to set assignees."""
+    is_editor = permissions.can_edit_feature(user, feature.key.integer_id())
+    approvers = approval_defs.get_approvers(gate.gate_type)
+    is_approver = permissions.can_review_gate(user, feature, gate, approvers)
+
+    if not is_editor and not is_approver:
+      self.abort(403, msg='User lacks permission to assign reviews')
+
+  def validate_assignees(self, assignees, fe, gate):
+    """Each assignee must have permission to review."""
+    approvers = approval_defs.get_approvers(gate.gate_type)
+    for email in assignees:
+      reviewer = User(email=email)
+      is_approver = permissions.can_review_gate(reviewer, fe, gate, approvers)
+      if not is_approver:
+        self.abort(400, 'Assignee is not a reviewer')
```

Comment:
nit: Maybe log the person who isn't  an assignee?

Reply:
All the POST data is logged in APIHandler.post() before this handler is called.